bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java

```diff
@@ -1466,7 +1490,11 @@ void recover(final GenericCallback<Void> cb) {
         if (wasInRecovery) {
             // if metadata is already in recover, dont try to write again,
             // just do the recovery from the starting point
-            new LedgerRecoveryOp(LedgerHandle.this, cb).initiate();
+            new LedgerRecoveryOp(LedgerHandle.this, cb)
+                    .parallelRead(bk.getConf().getEnableParallelRecoveryRead())
```

Comment:
Minor: reading from the AbstractConfiguration object for each entry can be expensive, since it goes to the HashTable and I think it convert from String into boolean even. If possible, cache the conf value into a final member variable int the constructor.